### DISCUSSION TITLE
Is there any documentation that explains why KFP v2 has 3 pods per component and what each pod does?

### DISCUSSION THREAD

**Discussion by droctothorpe** [kubeflow/pipelines#10865] [General]
This [design document](https://docs.google.com/document/d/1TZeZtxwPzAImIu8Jk_e-4otSx467Ckf0smNe7JbPReE/edit?usp=sharing&resourcekey=0-lTeZGW_Ys78j1LU60CEARg) may have some relevant info but it's not maintained. Why are there 3 pods per component? What does each pod do? Is there an easy to differentiate between the 3 types of pods so that users know which one to pull logs from? This is confusing for end users. It might be a good candidate for inclusion in the documentation page.

**Comment by HumairAK** [comment]
I don't think so, we should address it as part of: 

https://github.com/kubeflow/pipelines/issues/10957

**Comment by gregsheremeta** [comment]
Are there usually 3? I usually see 2 -- 1 is driver, 1 is the running task (wrapped in kfp-launcher)

**Reply by droctothorpe** [reply]
1 driver for the run as a whole, and 1 driver + 1 executor for each task.

**Reply by HumairAK** [reply]
yes that is correct, also note that in certain cases a new driver is added, for example when using the exit handler, this results in a new underlying argo wf DAG, which results in another driver pod (but no launcher pod)
